frontend/packages/dev-console/src/utils/resource-label-utils.ts

```diff
@@ -40,3 +42,15 @@ export const getPodLabels = (name: string) => {
     deploymentconfig: name,
   };
 };
+
+export const mergeData = (originalResource: K8sResourceKind, newResource: K8sResourceKind) => {
+  return {
+    ...originalResource,
+    ...newResource,
+    metadata: {
+      ...originalResource?.metadata,
+      ...newResource.metadata,
+      labels: newResource.metadata.labels,
+    },
+  };
```

Comment:
instead of this can we just have 

```
return {
    ...originalResource,
    ...newResource,
  };
```

Reply:
ignore above comment, I see for `imageStream` creation/update we get `422 Unprocessable Entity`

Reply:
@debsmita1 so we were using lodash merge since it recursively merges the old and new objects. Because we only want to change the old values with new and keep all the other data and this should happen recursively. The way you are adding the required metadata you will need to do the same thing for spec, template etc. since this way old spec will be replaced by new spec and the values that we needed from old spec will be erased. Better use lodash merge and only change update the labels.